### ISSUE TITLE
Support for Unity 5

### ISSUE THREAD

**Issue by astaerk** [astaerk/AutoRegistration#3]
As of now there is a new beta release of Unity 5.0
https://github.com/unitycontainer/unity/releases/tag/v5.0.0-beta

We should prepare support for Unity 5

**Comment by astaerk** [comment]
master is updated to unity 5.0.0-beta1

**Comment by astaerk** [comment]
master branch is updated to unity 5.0.0-beta1

**Comment by TRSynthro** [comment]
I just compiled my own version with Unity 5.1.
During testing I found an issue in Extension.GetAttribute(). 
It always returned null, because instead of GetCustomAttributes() the property CustomAttributes was used and this returns no Attribute. Therefore teh cas was always null.

The code below works for me.

        public static TAttr GetAttribute<TAttr>(this Type type)
            where TAttr : Attribute
        {
            return type.GetTypeInfo().GetCustomAttributes().Single(a => typeof(TAttr) == a.GetType()) as TAttr;
        }

**Comment by potatoestl** [comment]
I have been using unity auto registration and was attempting to upgrade to unity 5.8.6.
I was able to make the minimal changes to the current master to be able to build the solution.
Is AutoRegistration going to be updated and a new nuget package released as unity 5.x development seems to be finalized?

**Comment by astaerk** [comment]
@TRSynthro
First things first: Sorry for never giving you a reply. I just rechecked my mails and I never got notified about your comment. I had some problems with my mailserver around that time, so maybe it was that...
I will have a look at your problem and also at your proposal for a change.
Can you please tell me what .net version you are/were using?
Because I think the code you are talking about is simply there because I tried to support .net 4.0 as Unity now does. I'm still not sure if this is a good idea. :)

@potatoestl 
The plan was and still is to make updates and releases to the this project, yes. I'm just having some difficulties in getting the new csproj projects to build on the buildserver and I didn't want to release from my local machine. But since this has been a while since my last attempt I will give it another try the next in the next days.

**Comment by potatoestl** [comment]
It has been nearly two months and nothing.  We are going to release our project and would like to have the variance of one off packages removed before that happens.  Is my only way forward to pull the repo in-house and build a private package?

**Comment by astaerk** [comment]
Well, first of all I'm NOT doing this during my full time job!

But in the meantime I got a working build on the buildserver and I got my own projects ready to update to Unity 5. I had some work to do there without updating the dependencies.

Yesterday I was playing with the nuget config to understand how this works with version ranges and over the weekend I'm planning on testing all of this with a private NuGet feed.
I want to be a bit more assured, that the code works.
The feedback from TRSynthro unsetteled me a little bit if it will work or not.

But if it helps you I can at least release a beta package. Maybe you can give me some feedback then if it works for you or not.

**Comment by astaerk** [comment]
Ok, regarding the feedback from TRSynthro: I got it. I was always worried why I didn't have that problem, but I realized that I was using different code in my projects.
Over the weekend I reverted the change to using CustomAttributes back to using GetCustomAttributes, like it was before.
I had some trouble with the .Net Standard versions of unity regarding unit tests, but I think I sorted it out.

The plan now is to upload a package tomorrow in the evening.
